packages/contracts-ops/contracts/test/RebootVerify.t.sol

```diff
@@ -0,0 +1,256 @@
+    }
+
+    /// @notice Verify the state of Moonbeam
+    /// @dev The blockNumber is an arbitrary block number AFTER the deployment
+    function test_verifyMoonbeam() public {
+        domain = "moonbeam";
+        uint256 blockNumber = 2407153;
+        domainNumber = getDomainNumber(domain);
+        vm.createSelectFork(vm.envString("RPC_URL_MOONBEAM"), blockNumber);
+
+        verifyConfigImplementations();
+        readGovActions();
+        executeGovActions();
+        verifyGovActions();
+    }
+
+    /// @notice Verify the state of Milkomeda
+    /// @dev The blockNumber is an arbitrary block number AFTER the deployment
+    function test_milkomedaC1() public {
+        domain = "milkomedaC1";
+        uint256 blockNumber = 7355537;
+        domainNumber = getDomainNumber(domain);
+        vm.createSelectFork(vm.envString("RPC_URL_MILKOMEDAC1"), blockNumber);
+
+        verifyConfigImplementations();
+        readGovActions();
+        executeGovActions();
+        verifyGovActions();
+    }
+
+    /// @notice Verify that the implementations have been deployed at the addresses that exist
+    /// in the configuration file.
+    /// @dev To verify, we either call some constant that exists in the implementation, or a function that we
+    /// expect to exist
+    function verifyConfigImplementations() internal {
+        // verify home
+        homeImpl = Home(homeUpgrade(domain).implementation);
+        assertEq(homeImpl.MAX_MESSAGE_BODY_BYTES(), 2 * 2**10);
+        // verify replica
+        string memory remote = getConnections(domain)[0];
+        replicaImpl = Replica(replicaOfUpgrade(domain, remote).implementation);
+        assertEq(replicaImpl.LEGACY_STATUS_PROVEN(), bytes32(uint256(1)));
+        // verify gov router
+        govRouterImpl = GovernanceRouter(
+            governanceRouterUpgrade(domain).implementation
+        );
+        govRouterImpl.localDomain();
+        // verify bridge router
+        bridgeRouterImpl = BridgeRouter(
+            payable(bridgeRouterUpgrade(domain).implementation)
+        );
+        assertEq(bridgeRouterImpl.DUST_AMOUNT(), 0.06 ether);
+        // verify bridge token
+        bridgeTokenImpl = BridgeToken(
+            bridgeTokenUpgrade(domain).implementation
+        );
+        assertEq(
+            bridgeTokenImpl._PERMIT_TYPEHASH(),
+            keccak256(
+                "Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"
+            )
+        );
+        //verify token registry
+        tokenRegistryImpl = TokenRegistry(
+            tokenRegistryUpgrade(domain).implementation
+        );
+        tokenRegistryImpl.tokenBeacon();
+    }
+
+    /// @notice Verify the on-chain effects of executing the gov actions in the fork
+    function verifyGovActions() internal {
+        // Verify Updater rotation
+        UpdaterManager updaterManager = getUpdaterManager(domain);
+        assertEq(updaterManager.updater(), getUpdater(domain));
```

Comment:
could also assert it on the home

Reply:
also, this doesn't check replica updaters

Reply:
note: checking the replicas is probably the only thing about this PR i'd actually think good to change before merge, everything else is nice-to-have